### DISCUSSION TITLE
rec模型训练，单行文字，这样设计行不行，有没有什么问题？总字符4万，随机取出5个字符，另加一个短语 组成一行，这样8000行就能覆盖所有想训练的字符，8000*500=4百万就能均衡字符分布

### DISCUSSION THREAD

**Discussion by nissansz** [PaddlePaddle/PaddleOCR#12288] [Q&A]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：
- 版本号/Version：Paddle：  PaddleOCR：2.7.5 问题相关组件/Related components：
- 运行指令/Command Code：
- 完整报错/Complete Error Message：

rec模型训练，单行文字，这样设计行不行，有没有什么问题？总字符4万，随机取出5个字符，另加一个短语 组成一行，这样8000行就能覆盖所有想训练的字符，8000*500=4百万就能均衡字符分布

**Comment by UserWangZz** [comment]
可以详细说明一下思路吗

**Comment by nissansz** [comment]
就是想均衡字符分配，不知道怎么生成句子。
试验结果，发现英文空格都不见了，每次恢复训练，易错字容易出错。

![image](https://github.com/PaddlePaddle/PaddleOCR/assets/41010669/231bb42c-38ca-4882-932c-19e5127dc3eb)

**Comment by changdazhou** [comment]
建议在训练数据中补充一些空格试一下哈
